### PR TITLE
ci: fix storybook deployment actions

### PR DESCRIPTION
**Pull request by pregno**

### CODE REVIEW

File: .github/workflows/deploy-html-storybook-prod.yml [hunk 1]
```diff
@@ -82,8 +82,6 @@ jobs:
     name: Deploy to Amazon S3
     runs-on: ubuntu-latest
     needs: build
-    # Run if html is affected or manually triggered
-    if: ${{ needs.determine-affected.outputs.html_affected == 'true' || github.event_name == 'workflow_dispatch' }}
```
Comment (hamza14khan) [inline .github/workflows/deploy-html-storybook-prod.yml:86]: Dont we need this here for manual deployment? Happy to fix it for you.

File: .github/workflows/deploy-react-storybook-prod.yml [hunk 2]
```diff
@@ -9,8 +9,46 @@ env:
...
+            echo "react_affected=true" >> $GITHUB_OUTPUT
+          else
+            echo "react_affected=false" >> $GITHUB_OUTPUT
+          fi
+
   build:
     runs-on: ubuntu-latest
+    needs: determine-affected
+    # Run if html is affected or manually triggered
+    if: ${{ needs.determine-affected.outputs.react_affected == 'true' || github.event_name == 'workflow_dispatch' }}
```
Comment (hamza14khan) [inline .github/workflows/deploy-react-storybook-prod.yml:51]: Given that this check is here, we might need manual trigger here :D

File: .github/workflows/deploy-html-storybook-prod.yml (hunk 1 above)
Comment (pregno) [reply .github/workflows/deploy-html-storybook-prod.yml:86]: It depends on `build`, which has the same reference.
Plus is doesn't reference `determine-affected` so it wouldn't work anyway.

File: .github/workflows/deploy-react-storybook-prod.yml (hunk 2 above)
Comment (pregno) [reply .github/workflows/deploy-react-storybook-prod.yml:57]: thanks, forgot about it

**Review by hamza14khan — APPROVED** [review]
Great stuff 🥇

**Pull request merged**